### PR TITLE
Added hw_9.py

### PR DESCRIPTION
**Pull request by Oleksiy-Yurchenko**

### CODE REVIEW

File: hw_9.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+# Задача-2
+# Реализовать синглтон метакласс(класс для создания классов синглтонов).
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
```
Comment (katevoskresenska) [inline hw_9.py:40]: The new style is as follows: 
`cls._instances[cls] = super().__call__(*args, **kwargs)`

File: hw_9.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+        self.baskets.append(basket)
+        print('Basket with number ', basket.basket_id, ' was created.')
+        return basket
+
+    def find_basket(self, basket_id):
+        if self.baskets:
+            for basket in self.baskets:
+                if basket.basket_id == basket_id:
+                    return basket
+        return None
```
Comment (mmykhailenko) [inline hw_9.py:178]: These methods not for Warehouse class at all. They should be located in basket class

File: hw_9.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                          .format(material.name))
+            return order
+
+    def find_order(self, order_id):
+        if self.orders:
+            for order in self.orders:
+                if order.order_id == order_id:
+                    return order
+        return None
+
```
Comment (mmykhailenko) [inline hw_9.py:203]: same for order, they should be located in Order class or some class like OrderManager, but not in Warehouse

File: hw_9.py [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        if self.members:
+            for material in self.members:
+                id += 1
+                print(id, 'Material name: ', material.name, 'Material quantity :', material.quantity)
+
+
+basket_id = 0
+
+
+class Basket(Warehouse):
```
Comment (mmykhailenko) [inline hw_9.py:229]: Basket and Warehouse it's totally different classes. It's like a human and animal, both of them can walk or eat, but they can't be in relation parent and child.

File: hw_9.py [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+                id += 1
+                print(id, 'Material: ', material.name, 'Quantity: ', self.content[material], 'Price: ', material.price,
+                      'Total: ', self.content[material] *  material.price)
+            print('Grand Total: ', self.total)
+
+
+order_id = 0
+
+
+class Order(Basket):
```
Comment (mmykhailenko) [inline hw_9.py:262]: same here

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**